packages/typespec-ts/src/modular/helpers/clientHelpers.ts

```diff
@@ -62,25 +69,57 @@ export function getClientParameters(
     }
   }
 
-  const hasDefaultValue = (p: SdkParameter | SdkHttpParameter) =>
+  const hasDefaultValue = (
+    p:
+      | SdkMethodParameter
+      | SdkHttpParameter
+      | SdkEndpointParameter
+      | SdkCredentialParameter
+  ) =>
     p.clientDefaultValue || p.__raw?.defaultValue || p.type.kind === "constant";
-  const isRequired = (p: SdkParameter | SdkHttpParameter) =>
+  const isRequired = (
+    p:
+      | SdkMethodParameter
+      | SdkHttpParameter
+      | SdkEndpointParameter
+      | SdkCredentialParameter
+  ) =>
     !p.optional &&
     ((!hasDefaultValue(p) &&
       !(
-        p.type.kind === "endpoint" &&
-        p.type.templateArguments[0] &&
-        hasDefaultValue(p.type.templateArguments[0])
+        (p as any).type?.kind === "endpoint" &&
```

Comment:
remove any and Pls refer the pr https://github.com/Azure/typespec-azure/pull/2817 for TCGC migration guide.